internal/runner/runner_test.go

```diff
@@ -152,3 +250,59 @@ func TestCommandRunner_RunnerMulti(t *testing.T) {
 		t.Fatalf("unexpected env var value, want %q, got: %q", want, got)
 	}
 }
+
+func Test_dedupEnv(t *testing.T) {
+	tests := map[string]struct {
+		input    []string
+		expected []string
+	}{
+		"with duplicate": {
+			input: []string{
+				"FOO=hoge",
+				"FOO=bar",
+			},
+			expected: []string{
+				"FOO=bar",
+			},
+		},
+		"reverse order": {
+			input: []string{
+				"FOO=bar",
+				"FOO=hoge",
+			},
+			expected: []string{
+				"FOO=hoge",
+			},
+		},
+		"with extra": {
+			input: []string{
+				"FOO=hoge",
+				"EXTRA=fuga",
+				"FOO=bar",
+			},
+			expected: []string{
+				"FOO=bar",
+				"EXTRA=fuga",
+			},
+		},
+		"no duplicate": {
+			input: []string{
+				"FOO=bar",
+				"HOGE=fuga",
+			},
+			expected: []string{
+				"FOO=bar",
+				"HOGE=fuga",
+			},
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			sorted := cmpopts.SortSlices(func(a, b string) bool { return a < b })
```

Comment:
I shouldn't sort slices when comparing here. `dedupEnv` should output the vars in the original order they were in the list, removing the previous values in the list if they are overwritten.